### PR TITLE
wip support [time_area] with no specific timezone

### PR DESCRIPTION
**Pull request by gfgtdf**
now the c++ code does no longer add a dummy time when
there are no times defines for an [time_area]. this is
in particular useful when one wants to define areas in
the map editor that is then used ingame for exampel in
filters.
We should consider renamaing [time_area] to just [area]
[time_area]s that dont define their own times are now
ignored during tod calculation.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Looks nice! However I think it would be best if possible to rework the logic further so that `times_nonempty()` isn't needed.

**Comment by gfgtdf** [comment]
> Looks nice! However I think it would be best if possible to rework the logic further so that times_nonempty() isn't needed.

I could do that, but for that i'd need to figure out in particular how `calculate_best_liminal_bonus` is supposed to work. In fact i think the liminal calculation is quite broken  as it calculates some limial_bonus_ value  that neither considers [time_area]s nor is it updated when the time scedule changes and i didnt really want to turn this into a 'fix liminal' pr.

**Comment by CelticMinstrel** [comment]
> I could do that, but for that i'd need to figure out in particular how `calculate_best_liminal_bonus` is supposed to work.

If the issue is simple that the calculation requires at least one time, you could just make it return 0 without calculating when the schedule is empty. (Or would that break something?)

> In fact i think the liminal calculation is quite broken as it calculates some limial_bonus_ value that neither considers [time_area]s nor is it updated when the time scedule changes

This sounds like it could be something to look into, maybe we should open a new issue for it?

**Comment by GregoryLundberg** [comment]
Defining a named area sounds like a good idea, but should be a new PR.

The area could then be used as, for example, a filter condition sounds like a good idea, especially if the area can be a union of several smaller areas. Consider a weapon special which only works within the named area.

Even if the same effect can be achieved now, named areas could improve readability and maintainability.

If this PR is to fix a specific bug (sounds like it is), it should go in to snub the error, and "Fix time-of-day bonus/penalty calculations" sounds like it should be a new Issue.

**Comment by CelticMinstrel** [comment]
Um, @GregoryLundberg I think you're missing something.

Time areas are _already_ named areas. This PR just allows you to create one that doesn't have its own, separate schedule.

Defining an area as a union of other named areas should already be possible using `area=list,of,other,area,ids` in the area's filter.

File: src/scripting/game_lua_kernel.cpp [hunk 1]
```diff
@@ -3788,6 +3788,11 @@ int game_lua_kernel::intf_set_time_of_day(lua_State * L)
 			return luaL_argerror(L, 1, "invalid time of day ID");
 		}
 	}
+
+	if(new_time == 0 && num_times == 0) {
+		//ignore this case, becasue we dont want code like set_current_time(get_current_time()) to fail if num_times is 0.
```
Comment (GregoryLundberg) [inline src/scripting/game_lua_kernel.cpp]: spelling: because

**Pull request merged**